File: stats_series_01_week_14.R
# Snippet 1: Support Vector Machines and Kernel Trick
library(e1071)
svc_linear <- svm(y ~ ., data=trainData, kernel="linear")
svc_rbf <- svm(y ~ ., data=trainData, kernel="radial")

# Snippet 2: Ensemble Learning Methods
library(ada)
library(gbm)
ada <- ada(y ~ ., data=trainData)
gboost <- gbm(y ~ ., data=trainData, distribution="gaussian")

# Snippet 3: Neural Networks and Deep Learning
library(keras)
model <- keras_model_sequential()
model %>% layer_dense(units=10, activation='relu') %>% layer_dense(units=1, activation='sigmoid')
model %>% compile(loss='binary_crossentropy', optimizer='adam')
model %>% fit(X_train, y_train, epochs=10)

# Snippet 4: Reinforcement Learning (conceptual introduction as Q-learning is more involved in R)

File: stats_series_01_week_14.py
# Snippet 1: Support Vector Machines and Kernel Trick
from sklearn.svm import SVC
svc_linear = SVC(kernel='linear').fit(X_train, y_train)
svc_rbf = SVC(kernel='rbf').fit(X_train, y_train)

# Snippet 2: Ensemble Learning Methods
from sklearn.ensemble import AdaBoostClassifier, GradientBoostingClassifier
ada = AdaBoostClassifier().fit(X_train, y_train)
gboost = GradientBoostingClassifier().fit(X_train, y_train)

# Snippet 3: Neural Networks and Deep Learning
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
model = Sequential([Dense(10, activation='relu'), Dense(1, activation='sigmoid')])
model.compile(loss='binary_crossentropy', optimizer='adam')
model.fit(X_train, y_train, epochs=10)

# Snippet 4: Reinforcement Learning (simple Q-learning example)
import numpy as np
Q = np.zeros([state_space, action_space])
for episode in range(1,1001):
    state = env.reset()
    # ... Q-learning logic ...

